Packs/TrendMicroEmailSecurity/Integrations/TrendMicroEmailSecurityEventCollector/TrendMicroEmailSecurityEventCollector.py

```diff
@@ -0,0 +1,488 @@
+        pass
+
+    return "ok"
+
+
+def fetch_events_command(
+    client: Client,
+    args: dict[str, str],
+    first_fetch: str,
+    last_run: dict,
+) -> tuple[list[dict], dict]:
+    """
+    Args:
+        client (Client): The client for api calls.
+        args (dict[str, str]): The args.
+        first_fetch (str): The first fetch time.
+        last_run (dict): The last run dict.
+
+    Returns:
+        tuple[list[dict], dict]: List of all event logs of all types,
+                                 The updated `last_run` obj.
+    """
+    hide_sensitive = argToBoolean(args.get("hide_sensitive", True))
+    time_to = datetime.now()
+    limit: int = arg_to_number(args.get("max_fetch", "5000"))  # type: ignore[assignment]
+
+    events: list[dict] = []
+    new_last_run: dict[str, str] = {}
+    for event_type in EventType:
+        time_from = last_run.get(f"time_{event_type.value}_from") or first_fetch
+        ids_fetched_by_type = last_run.get(
+            f"fetched_event_ids_of_{event_type.value}", []
+        )
+
+        events_by_type, deduplicate_management = fetch_by_event_type(
+            client=client,
+            start=time_from,
+            end=time_to.strftime(DATE_FORMAT_EVENT),
+            limit=limit,
+            ids_fetched_by_type=ids_fetched_by_type,
+            event_type=event_type,
+            hide_sensitive=hide_sensitive,
+        )
+
+        events.extend(events_by_type)
+
+        last_run_for_type = managing_set_last_run(
+            events=events_by_type,
+            last_run=last_run,
+            start=time_from,
+            event_type=event_type,
+            deduplicate=deduplicate_management,
+        )
+        new_last_run.update(last_run_for_type)
+    demisto.debug(f"Done fetching, got {len(events)} events.")
+
+    return events, new_last_run
+
+
+def main() -> None:  # pragma: no cover
+    params = demisto.params()
+    args = demisto.args()
+
+    base_url = params["url"].strip("/")
+    username = params["credentials"]["identifier"]
+    api_key = params["credentials"]["password"]
+    verify = not params.get("insecure", False)
+    proxy = params.get("proxy", False)
+    first_fetch = set_first_fetch()
+
+    should_push_events = argToBoolean(args.get("should_push_events", False))
+
+    command = demisto.command()
+    try:
+        client = Client(
+            base_url=base_url,
+            username=username,
+            api_key=api_key,
+            verify=verify,
+            proxy=proxy,
+        )
+
+        if command == "test-module":
+            return_results(test_module(client))
+
+        elif command == "trend-micro-get-events":
+            should_update_last_run = False
+            since = set_first_fetch(args.get("since") or "1 days")
+            events, _ = fetch_events_command(client, args, since, last_run={})
+            return_results(
+                CommandResults(readable_output=tableToMarkdown("Events:", events))
```

Comment:
do we not return them to context? 🤔

Reply:
No, there is an option to set the `should_push_events` argument to true, then the logs will be returned to the dataset

Reply:
then please add a comment explaining that, for future maintainers